### PR TITLE
ci: fix and test with Wunused-macros

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6937

Describe changes:
- Remove unused macros as reported by compiler with `-Wunused-macros` and test in CI with default C flags

#10804 which should be good for CI

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | build_asan | | |
 

Pipeline 20015

File: src/source-af-packet.c [hunk 1]
```diff
@@ -77,19 +76,6 @@
 #include <bpf/bpf.h>
 #endif
 
-struct bpf_program {
-    unsigned int bf_len;
-    struct bpf_insn *bf_insns;
-};
```
Comment (jufajardini) [inline src/source-af-packet.c:83]: Seems that we need this one? https://github.com/OISF/suricata/actions/runs/8652209800/job/23724605678#step:12:54

Comment (catenacyber) [reply src/source-af-packet.c:83]: We should not, but this is include hell :
`bpf_program` may be defined by both kernel bpf.h and by libpcap pcap.h (and other like bpf_insn)
So, we should only be using either pcap or bpf in each C file, but this is not the case here

Comment (catenacyber) [reply src/source-af-packet.c:83]: bpf.h is needed for `bpf_map_update_elem(mapd, key, value, BPF_NOEXIST)`

pcap.h is needed for `PCAP_ERRBUF_SIZE` and `SCBPFCompile`

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/10814

**Pull request closed without merging**